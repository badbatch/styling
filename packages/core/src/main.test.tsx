import { mount } from "enzyme";
import React from "react";
import styling from "./main";

describe("styling", () => {
  const PrimaryButton = styling(
    "button",
    [
      "block",
      "inverse",
      "disabled",
      "error",
      "externalText",
      "noText",
      ["size", ["xs", "sm", "md"]],
      "stretch",
      ["variant", ["primary", "secondary", "link"]],
    ],
    [
      "block",
      "disabled",
      "error",
      "externalText",
      "inverse",
      "noText",
      "stretch",
      "variant::link",
      "size::xs",
      "size::sm",
    ],
    {
      base: "button__primary-button-1842608486",
      block: "button__primary-button-1842608486--block",
      disabled: "button__primary-button-1842608486--disabled",
      error: "button__primary-button-1842608486--error",
      "external-text": "button__primary-button-1842608486--external-text",
      inverse: "button__primary-button-1842608486--inverse",
      "no-text": "button__primary-button-1842608486--no-text",
      "no-text--size-sm": "button__primary-button-1842608486--no-text--size-sm",
      "no-text--size-xs": "button__primary-button-1842608486--no-text--size-xs",
      stretch: "button__primary-button-1842608486--stretch",
      "variant-link": "button__primary-button-1842608486--variant-link",
    },
  );

  describe("WHEN a tag name is passed in as the component", () => {
    it("SHOULD return the correct component with the correct class names", () => {
      expect(mount(<PrimaryButton noText size="sm" />).html()).toMatchSnapshot();
    });
  });

  describe("WHEN a React component is passed in as the component", () => {
    const PrimaryButtonStyledSpan = styling(
      PrimaryButton,
      [
        "block",
        "inverse",
        "disabled",
        "error",
        "externalText",
        "noText",
        ["size", ["xs", "sm", "md"]],
        "stretch",
        ["variant", ["primary", "secondary", "link"]],
      ],
      [],
      {
        base: "button__primary-button-styled-span-759899086",
      },
    );

    it("SHOULD return the correct component with the correct class names", () => {
      expect(mount(<PrimaryButtonStyledSpan noText size="sm" />).html()).toMatchSnapshot();
    });
  });
});
