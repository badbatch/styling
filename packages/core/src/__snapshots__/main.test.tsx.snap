// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styling WHEN a React component is passed in as the component SHOULD return the correct component with the correct class names 1`] = `"<button class=\\"button__primary-button-1842608486 button__primary-button-1842608486--no-text--size-sm button__primary-button-1842608486--no-text button__primary-button-styled-span-759899086\\" data-styling-classnames=\\"button__primary-button-styled-span-759899086\\"></button>"`;

exports[`styling WHEN a tag name is passed in as the component SHOULD return the correct component with the correct class names 1`] = `"<button class=\\"button__primary-button-1842608486 button__primary-button-1842608486--no-text--size-sm button__primary-button-1842608486--no-text\\" data-styling-classnames=\\"button__primary-button-1842608486 button__primary-button-1842608486--no-text--size-sm button__primary-button-1842608486--no-text\\"></button>"`;
